File: spikes/telegram-engine/adapters/teleproto/create_session.mjs
import readline from "node:readline/promises";
import { stdin as input, stdout as output } from "node:process";
import { TelegramClient } from "teleproto";
import { StringSession } from "teleproto/sessions/index.js";

function requiredEnv(name) {
  const value = (process.env[name] ?? "").trim();
  if (!value) throw new Error(`Missing environment variable: ${name}`);
  return value;
}

const rl = readline.createInterface({ input, output });
const apiId = Number(requiredEnv("TELEGRAM_TEST_API_ID"));
const apiHash = requiredEnv("TELEGRAM_TEST_API_HASH");
const twoFactorPassword = process.env.TELEPROTO_2FA_PASSWORD ?? "";
const client = new TelegramClient(new StringSession(""), apiId, apiHash, {
  connectionRetries: 3,
});

try {
  await client.start({
    phoneNumber: () => rl.question("Nomor Telegram (+62...): "),
    phoneCode: () => rl.question("OTP Telegram: "),
    password: async () => twoFactorPassword,
    onError: (error) => {
      console.error(`Login gagal: ${error.message}`);
      return true;
    },
  });

  console.log("\nSALIN BARIS INI KE spikes/telegram-engine/.env:");
  console.log(`TELEPROTO_TEST_SESSION=${client.session.save()}`);
} finally {
  rl.close();
  await client.disconnect();
}

File: spikes/telegram-engine/adapters/telethon/create_session.py
"""Create a Telethon StringSession locally without persisting credentials."""

import asyncio
import os
from getpass import getpass

from telethon import TelegramClient
from telethon.sessions import StringSession


def required_env(name: str) -> str:
    value = os.environ.get(name, "").strip()
    if not value:
        raise SystemExit(f"Missing environment variable: {name}")
    return value


async def main() -> None:
    api_id = required_env("TELEGRAM_TEST_API_ID")
    api_hash = required_env("TELEGRAM_TEST_API_HASH")

    client = TelegramClient(StringSession(), int(api_id), api_hash)
    try:
        await client.start(
            phone=lambda: input("Nomor Telegram (+62...): "),
            code_callback=lambda: input("OTP Telegram: "),
            password=lambda: getpass("Password 2FA: "),
        )
        print("\nSALIN BARIS INI KE spikes/telegram-engine/.env:")
        print("TELETHON_TEST_SESSION=" + client.session.save())
    finally:
        await client.disconnect()


if __name__ == "__main__":
    asyncio.run(main())
